include/boost/gil/image_processing/filter.hpp

```diff
@@ -135,6 +137,64 @@ void median_filter(SrcView const& src_view, DstView const& dst_view, std::size_t
     }
 }
 
+namespace detail {
+
+template<typename KernelT>
+void getGaussianKernel(KernelT& kernel,
```

Comment:
You may wanna have a look at the already implemented gaussian kernel generator at:
https://github.com/boostorg/gil/blob/f33128e20edd676728f826107adac19c22be9651/include/boost/gil/image_processing/numeric.hpp#L129 

Please consider using it. This may induce significant change in the code

Reply:
Yeah, I had a look over it. But it returns a 2D kernel. But Gaussian Kernel has a separable property hence 2D convolution is broken down into 2 passes of 1D convolution. Plz, correct me if I'm wrong:)

Reply:
@laxsuryavanshi Please, follow naming used in the existing code. Any form of CamelCase for functions and classes is no-go, i.e. `getGaussianKernel` must read `get_gaussian_kernel`.

> But Gaussian Kernel has a separable property hence 2D

We could have utility converting our kernels 1D<->2D

Reply:
> We could have utility converting our kernels 1D<->2D
This is a nice idea, will work on it

Reply:
> > We could have utility converting our kernels 1D<->2D
> 
> This is a nice idea, will work on it

@laxsuryavanshi I forgot to mention, if you aim to add such conversion functions, please submit it as a separate PR (including at least minimal tests). Thanks!